modules/sdk-coin-ton/src/lib/transaction.ts

```diff
@@ -67,8 +67,8 @@ export class Transaction extends BaseTransaction {
     return Buffer.from(this.unsignedMessage, 'hex');
   }
 
-  async build(): Promise<void> {
-    this._type = TransactionType.Send;
+  async build(txType: TransactionType): Promise<void> {
```

Comment:
you shouldn't change the build method signature here, the type should be from this._type